packages/instantsearch-ui-components/src/components/Hits.tsx

```diff
@@ -8,6 +8,54 @@ type Hit = Record<string, unknown> & {
   objectID: string;
 };
 type SendEventForHits = (...props: unknown[]) => void;
+type Banner = {
+  image: {
+    urls: Array<{
+      url: string;
+    }>;
+    title?: string;
+  };
+  link?: {
+    url?: string;
+    target?: '_blank' | '_self';
+  };
+};
+
+type BannerProps = ComponentProps<'aside'> & {
+  banner: Banner;
+  classNames: Pick<
+    Partial<HitsClassNames>,
+    'bannerRoot' | 'bannerLink' | 'bannerImage'
+  >;
+};
+
+function createDefaultBannerComponent({ createElement }: Renderer) {
+  return function DefaultBanner({ classNames, banner }: BannerProps) {
+    return (
+      <aside className={cx('ais-Hits-banner', classNames.bannerRoot)}>
+        {banner.link?.url ? (
+          <a
+            className={cx('ais-Hits-banner-link', classNames.bannerLink)}
+            href={banner.link.url}
+            target={banner.link.target}
+          >
+            <img
+              className={cx('ais-Hits-banner-image', classNames.bannerImage)}
+              src={banner.image.urls[0].url}
+              alt={banner.image.title}
+            />
+          </a>
+        ) : (
+          <img
+            className={cx('ais-Hits-banner-image', classNames.bannerImage)}
+            src={banner.image.urls[0].url}
```

Comment:
if `image.urls[0]` is undefined, this display and the surrounding components don't really make sense. Maybe it should early return `null` in that case? Alternatively even in the condition where it's chosen to render the banner at all.

Reply:
`image.urls[0]` should never be undefined based on the typing (because you're correct - a banner without an image would not make sense). And `DefaultBannerComponent` should only render if the banner data is not undefined.

Do you think we need other guardrails in place here?